### ISSUE TITLE
Issue !TARGET_OS_TV React Native, cannot compile to iOS, and i have remove this code to make it works.

### ISSUE THREAD

**Issue by primayudantra** [facebook/react-native#12746]
I got this error : 
react-native/React/Profiler/RCTProfile.m:386:57: Incompatible block pointer types assigning to 'UIActivityViewControllerCompletionWithItemsHandler _Nullable' (aka 'void (^)(NSString * _Nullable __strong, BOOL, NSArray * _Nullable __strong, NSError * _Nullable __strong)') from 'int ((^)(void))'

when i compiled with XCODE i got this error and i must remove this code.

```
#if !TARGET_OS_TV
      UIActivityViewController *activityViewController = [[UIActivityViewController alloc] initWithActivityItems:@[[NSURL fileURLWithPath:outFile]]
                                                                                           applicationActivities:nil];
      activityViewController.completionWithItemsHandler = ^(__unused UIActivityType activityType,
                                                            __unused BOOL completed,
                                                            __unused NSArray *items,
                                                            __unused NSError *error) {
        RCTProfileControlsWindow.hidden = NO;
      };
      RCTProfileControlsWindow.hidden = YES;
      dispatch_async(dispatch_get_main_queue(), ^{
        [[[[[UIApplication sharedApplication] delegate] window] rootViewController] presentViewController:activityViewController
                                                                                                 animated:YES
                                                                                               completion:nil];
      });

#endif
```

why?

**Comment by hramos** [comment]
This issue looks like a question that would be best asked on [StackOverflow](http://stackoverflow.com/questions/tagged/react-native).

StackOverflow is amazing for Q&A: it has a reputation system, voting, the ability to mark a question as answered. Because of the reputation system it is likely the community will see and answer your question there. This also helps us use the GitHub bug tracker for bugs only.

Will close this as this is really a question that should be asked on StackOverflow.

**Comment by ekimlinger** [comment]
I don't believe that this is an issue with his system alone. I too was receiving the same error today when attempting to run. @primayudantra Were you ever able to figure this out?

**Comment by hramos** [comment]
Feel free to open a new issue if you believe there is something actionable that can be solved with a PR. Can you make sure to fill out the template if you do so?

**Comment by mhdayyoob** [comment]
I also get this error when i create a new project

**Comment by kesjien** [comment]
same for me

**Comment by tk1cntt** [comment]
+1 same issue

**Comment by mhdayyoob** [comment]
any Updates for the issue

**Comment by hramos** [comment]
This issue is closed and metoo comments aren't helpful. I suggest opening a new issue, making sure to fill out the template with all the required information.

**Comment by disparu86** [comment]
+1 same

**Comment by akashdmahali** [comment]
@hramos the issue is within the react native source code as can be seen from the source file path "react-native/React/Profiler/RCTProfile.m" . I too faced this issue. 

After digging bit deeper, it seems in RN version 0.42.0 a change was made at the exact same location where this issue is occurring. Here is the commit for your reference. [GitHub commit](https://github.com/facebook/react-native/commit/453e70d19c28b56dfcecfe205ccd4ccad2b33009). Moreover this started occurring only after upgrading to version 0.42.0. 

As @primayudantra  mentioned , after commenting out the said code, the Xcode project complied and ran properly, however changing the source files of RN is definitely not a solution here.

Also i do understand that this issue has been closed, but since many people are facing this issue, I wanted to share my view here. @hramos we would appreciate any inputs from your side.

**Comment by hramos** [comment]
Thanks for opening a new issue with all the necessary information.
